bare_metal_billing/main.py

```diff
@@ -107,6 +112,11 @@ def main():
     logger.info(f"Interval for processing {args.start} - {args.end}.")
     logger.info(f"Invoice file will be saved to {args.output_file}.")
 
+    # Default to fetching lease file one day before billing end date because end date is not-inclusive
+    bm_usage_file = args.bm_usage_file or s3_bucket.fetch_s3(
+        f"{args.invoice_month}/esi-lease-{(default_end_argument() - timedelta(days=1)).strftime("%Y%m%d")}.json"
+    )
```

Comment:
I know you like using `or` in these cases but it doesn't read well in this case because eight now it is

`variable_x = something or something-else-long-and-multi-line` 

The line on 117 is also quite long also.

I'd rather

```python
if args.bm_usage_file:
    bm_usage_file = args.bm_usage_file
else:
    previous_day = (default_end_argument() - timedelta(days=1)).strftime("%Y%m%d")
    bm_usage_file = s3_bucket.fetch_s3(f"{args.invoice_month}/esi-lease-{previos_day}.json")
```

This reads much better to me. This is purely a stylistic choice in my opinion so feel free to disregard this.